### PR TITLE
Fix HTTP response body not being closed after reading

### PR DESCRIPTION
**Pull request by faddat**
Closes: #1913

Ensures that we close a response body in a test that would frequently fail. 

---

#### PR checklist

- [x] Tests written/updated
- [ ] Changelog entry added in `.changelog` (we use [unclog](https://github.com/informalsystems/unclog) to manage our changelog)
- [ ] Updated relevant documentation (`docs/` or `spec/`) and code comments

### CODE REVIEW

File: rpc/jsonrpc/server/http_server_test.go [hunk 1]
```diff
@@ -144,10 +144,15 @@ func TestWriteRPCResponseHTTPError(t *testing.T) {
...
 		types.RPCInternalError(types.JSONRPCIntID(-1), errors.New("foo")))
 	require.NoError(t, err)
+
 	resp := w.Result()
 	body, err := io.ReadAll(resp.Body)
-	_ = resp.Body.Close()
 	require.NoError(t, err)
+
+	// Ensure the response body is closed after reading
+	defer resp.Body.Close()
```
Comment (melekes) [inline rpc/jsonrpc/server/http_server_test.go:153]:
```suggestion
	err = resp.Body.Close()
	require.NoError(t, err)
```

File: rpc/jsonrpc/server/http_server_test.go [hunk 2]
```diff
@@ -144,10 +144,15 @@ func TestWriteRPCResponseHTTPError(t *testing.T) {
 		http.StatusInternalServerError,
 		types.RPCInternalError(types.JSONRPCIntID(-1), errors.New("foo")))
 	require.NoError(t, err)
+
 	resp := w.Result()
 	body, err := io.ReadAll(resp.Body)
-	_ = resp.Body.Close()
 	require.NoError(t, err)
+
+	// Ensure the response body is closed after reading
```
Comment (melekes) [inline rpc/jsonrpc/server/http_server_test.go:152]:
```suggestion
```

File: rpc/jsonrpc/server/http_server_test.go [hunk 3]
```diff
@@ -144,10 +144,15 @@ func TestWriteRPCResponseHTTPError(t *testing.T) {
...
+	defer resp.Body.Close()
+
+	// Assertions
```
Comment (melekes) [inline rpc/jsonrpc/server/http_server_test.go:155]:
```suggestion
```

File: rpc/jsonrpc/server/http_server_test.go [hunk 4]
```diff
@@ -144,10 +144,15 @@ func TestWriteRPCResponseHTTPError(t *testing.T) {
 		http.StatusInternalServerError,
 		types.RPCInternalError(types.JSONRPCIntID(-1), errors.New("foo")))
 	require.NoError(t, err)
+
 	resp := w.Result()
 	body, err := io.ReadAll(resp.Body)
-	_ = resp.Body.Close()
 	require.NoError(t, err)
+
```
Comment (melekes) [inline rpc/jsonrpc/server/http_server_test.go:151]:
```suggestion
```

**Review by melekes — COMMENTED** [review]
Thanks @faddat ❤️ 

I don't see a need to `defer` closing the body. Am I missing something?

**Comment by faddat** [comment]
@melekes -- For whatever reason, the `defer` did seem to make a difference but I was a bit head scratchy like yourself.  We could just hold off merging any of these until

* #1954 

goes green a frew hundred times?

**Review by melekes — APPROVED** [review]
👍

**Pull request merged**